board.cpp

```diff
@@ -0,0 +1,164 @@
+#include "board.h"
+
+bool isPositionValid(Board board, Position pos){ //check if pos if withing acceptable values 
+    if(pos.xpos < 0 || pos.ypos < 0) return false;
+    if(pos.xpos > board.width || pos.ypos > board.height) return false;
+    for(std::vector<Position>::iterator it = board.disabledPositions.begin(); it != board.disabledPositions.end(); it++){ //check if pos is not equal to an invalid position
+        if(*it == pos){
+            return false;
+        }
+    }
+    return true;
+}
+
+//gets all possible valid positions
+//mainly for bot logic
+std::vector<Position> getValidTiles(Board board){ 
+    std::vector<Position> validPositions;
+    for(int x = 1; x <= board.width; x++){
+        for(int y = 1; y <= board.height; y++){
+            Position pos;
+            pos.xpos = x;
+            pos.ypos = y;
+            if (isPositionValid(board, pos)){
+                validPositions.push_back(pos);
+            }
+        }
+    }
+    return validPositions;
+}
+
+//board construction
+Board createBoard(){
+    Board board;
+    bool created = false;
+    int width = NULL;
+    int height = NULL;
+    int mineCount = 0;
+    std::string pvx = "";
+    while (!(pvx == "pve") && !(pvx == "pvp")){
+        std::cout << "will this be a player vs player game or a player vs AI game?" << std::endl;
+        std::cout << "write 'pvp' for player vs player or 'pve' for player vs AI" << std::endl;
+        std::cin >> pvx;
+    }
+    while ((5 > width) || (width > 10)){
+        std::cout << "choose the width of the field (between 5 and 10)" << std::endl;
+        std::cin >> width;
+    }
+    while ((5 > height) || (height > 10)){
+        std::cout << "choose the height of the field (between 5 and 10)" << std::endl;
+        std::cin >> height;
+    }
+    while ((3 > mineCount) || (mineCount > 8)){
+        std::cout << "choose the number of mines on the field (between 3 and 8)" << std::endl;
+        std::cin >> mineCount;
+    }

```

Comment:
#should same as above, move outside board creation function and pass values as parameters instead.